web/src/test/java/com/graphhopper/http/resources/RouteResourceClientHCTest.java

```diff
@@ -0,0 +1,370 @@
+        GHRequest req = new GHRequest().
+                addPoint(new GHPoint(42.507065, 1.529846)).
+                addPoint(new GHPoint(42.510383, 1.533392)).
+                putHint("vehicle", "car");
+
+        GHResponse res = gh.route(req);
+        InstructionList instructions = res.getBest().getInstructions();
+        String finishInstructionName = instructions.get(instructions.size() - 1).getName();
+        assertEquals("Arrive at destination", finishInstructionName);
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME)
+    @CsvSource({"true,0", "true,1000", "false,-1"})
+    public void doNotReadFinishInstruction(boolean usePost, int maxUnzippedLength) {
+        GraphHopperWeb gh = createGH(usePost, maxUnzippedLength);
+        GHRequest req = new GHRequest().
+                addPoint(new GHPoint(42.507065, 1.529846)).
+                addPoint(new GHPoint(42.510383, 1.533392)).
+                putHint("vehicle", "car").
+                putHint("turn_description", false);
+        GHResponse res = gh.route(req);
+        InstructionList instructions = res.getBest().getInstructions();
+        String finishInstructionName = instructions.get(instructions.size() - 1).getName();
+        assertEquals("", finishInstructionName);
+    }
+
+    void isBetween(double from, double to, double expected) {
+        assertTrue(expected >= from, "expected value " + expected + " was smaller than limit " + from);
+        assertTrue(expected <= to, "expected value " + expected + " was bigger than limit " + to);
+    }
+
+    @Test
+    public void testUnknownInstructionSign() throws IOException {
+        // Modified the sign though
+        ObjectMapper objectMapper = Jackson.newObjectMapper();
+        JsonNode json = objectMapper.readTree("{\"instructions\":[{\"distance\":1.073,\"sign\":741,\"interval\":[0,1],\"text\":\"Continue onto A 81\",\"time\":32,\"street_name\":\"A 81\"},{\"distance\":0,\"sign\":4,\"interval\":[1,1],\"text\":\"Finish!\",\"time\":0,\"street_name\":\"\"}],\"descend\":0,\"ascend\":0,\"distance\":1.073,\"bbox\":[8.676286,48.354446,8.676297,48.354453],\"weight\":0.032179,\"time\":32,\"points_encoded\":true,\"points\":\"gfcfHwq}s@}c~AAA?\",\"snapped_waypoints\":\"gfcfHwq}s@}c~AAA?\"}");
+        PathWrapper wrapper = PathWrapperDeserializer.createPathWrapper(objectMapper, json, true, true);
+
+        assertEquals(741, wrapper.getInstructions().get(0).getSign());
+        assertEquals("Continue onto A 81", wrapper.getInstructions().get(0).getName());
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME)
+    @CsvSource({"true,0", "true,1000", "false,-1"})
+    public void testPathDetails(boolean usePost, int maxUnzippedLength) {
+        GraphHopperWeb gh = createGH(usePost, maxUnzippedLength);
+        GHRequest req = new GHRequest().
+                addPoint(new GHPoint(42.507065, 1.529846)).
+                addPoint(new GHPoint(42.510383, 1.533392)).
+                putHint("profile", "my_car");
```

Comment:
You need to use `setProfile()` here. For POST `putHint` also seems to work (because of json serialization/deserialization probably). Ugh this is a bit ugly.

Reply:
Ah, yeah.

Reply:
We could fix this when we internally use the hints Map for "profile". (btw: we would have the same problem with "algorithm" and probably all the other properties)

Reply:
Why is there a problem with "algorithm" and the others? The problem here seems that we are calling `putHint` instead of `setProfile`? Or do you think its better to be able to use both `putHint` and `setProfile`? Or do you want to remove `setProfile` altogether? I have no strong opinion so far, but since `profile` is even more or less required it seems to deserve its own setter? And if it has a setter it should also have its own field (because Jackson expects this by convention).

Reply:
> The problem here seems that we are calling putHint instead of setProfile?

But for GET this approach works and so people might do it? (not sure)

> Or do you want to remove setProfile altogether?

No, just put the profile string into hints under the hood like
```java
... setProfile(String profile) {
hints.put("profile", profile);
return this;
}
```

Not sure if this is nice ...

Reply:
> But for GET this approach works and so people might do it? (not sure)

I think it only works for POST, but you are right it should better work the same way for both. If this is our only concern we could also adjust `GraphHopperWeb#putHint` to set the profile if the given key is `profile`? Btw what do you think of removing GET from client-hc altogether? I do not see how it can be useful (except for writing tests that cover both GET and POST maybe but that is not really the purpose of client-hc :)). 

> Not sure if this is nice ...

Yes me neither. I think when there is `setProfile` Jackson assumes there is a profile field, but not sure if we need to care about this here...

Reply:
> Btw what do you think of removing GET from client-hc altogether?

Hmmh, sounds good. Initially I tought we need a certain transition phase to make the POST request more stable, but now everything seems to work already.

Reply:
Let's do this in another issue :) ?

Reply:
> but now everything seems to work already

We should also fix #1996 first then (for example we still have #1995). The only thing I am wondering is if we give POST the higher priority why would you remove gpx export (#2006)? Is this feature somehow not useful anymore or replaced by something else or only supported for GH maps?

Reply:
While the purpose of client-hc is certainly not testing GET&POST /route we still need some way to do this efficiently, it is quite annoying to test both endpoints separately and having bugs in one but not the other endpoint and so on. Or we manage to do very little in the Resource such that there is not much room for errors...